core/src/avm2/object/stage_object.rs

```diff
@@ -124,8 +125,9 @@ impl<'gc> TObject<'gc> for StageObject<'gc> {
         self.0.read().display_object
     }
 
-    fn init_display_object(&self, mc: MutationContext<'gc, '_>, obj: DisplayObject<'gc>) {
+    fn init_display_object(&self, mc: MutationContext<'gc, '_>, mut obj: DisplayObject<'gc>) {
         self.0.write(mc).display_object = Some(obj);
+        obj.set_object2(mc, (*self).into());
```

Comment:
Thanks for fixing this :+1: 
Now that it's here, I think we can remove the `set_object2` call from simplebutton.rs and displayobject.rs .